src/workerd/api/actor-state.c++

```diff
@@ -889,8 +889,14 @@ void DurableObjectState::abort(jsg::Optional<kj::String> reason) {
     s.get()->getActorCacheInterface().shutdown(error);
   }
 
-  IoContext::current().abort(kj::cp(error));
-  kj::throwFatalException(kj::mv(error));
+  IoContext::current().abort(kj::mv(error));
+  js.terminateExecution();
+
+  // terminateExecution() only sets a flag. In order for V8 to notify it, we have to make it do
+  // some work that causes it to check the flag. This has been observed to do the trick.
+  // TODO(cleanup): This appears in ExecutionContext::abort(), DurableObjectState::abort(), and
+  //     processExitImpl() in node/util.c++. Consolidate?
+  jsg::check(v8::JSON::Stringify(js.v8Context(), js.str()));
```

Comment:
Hmm.. this is a bit ugly. I wonder if a variation on `js.terminateExecution()` could encapsulate this, e.g. `js.terminateExecution({ force: true });` or something similar.

Reply:
OK I added a commit to clean this up: 6f4aa13084423f3f745cd29a514fed5a6847c6f8